x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts

```diff
@@ -104,6 +115,17 @@ describe('EnterpriseSearchRequestHandler', () => {
           'http://localhost:3002/api/example?someQuery=true'
         );
       });
+
+      it('will correctly encode query string parameters', async () => {
+        const requestHandler = enterpriseSearchRequestHandler.createRequest({
+          path: '/api/example',
+        });
+        await makeAPICall(requestHandler, { query: { 'page[current]': 1 } });
+
+        EnterpriseSearchAPI.shouldHaveBeenCalledWith(
+          'http://localhost:3002/api/example?page%5Bcurrent%5D=1'
+        );
+      });
```

Comment:
Per above, if we want to include URI encoded paths in our tests
```suggestion
      it('correctly encodes paths and query string parameters', async () => {
        const requestHandler = enterpriseSearchRequestHandler.createRequest({
          path: '/api/some example',
        });
        await makeAPICall(requestHandler, { query: { 'page[current]': 1 } });

        EnterpriseSearchAPI.shouldHaveBeenCalledWith(
          'http://localhost:3002/api/some%20example?page%5Bcurrent%5D=1'
        );
      });
```